src/pathfinding_dijikstra.h

```diff
@@ -0,0 +1,254 @@
+#include <algorithm>
+#include <functional>
+#include <optional>
+#include <vector>
+
+#include "cata_utility.h"
+#include "game_constants.h"
+#include "map.h"
+#include "point.h"
+#include "rng.h"
+
+const size_t DIJIKSTRA_ARRAY_SIZE = OVERMAP_LAYERS * MAPSIZE_Y * MAPSIZE_X;
+typedef std::pair<float, tripoint> val_pair;
+
+// A struct defining abilities of the actor and how to respond to various terrain features
+struct PathfindingSettings {
+    // Our bash strength
+    int bash_strength = 0;
+    // Even if we can bash, multiply time needed to do it by this
+    // >1 to make bashes less attractive, <1 to make them more attractive. Do not use negative values.
+    float bash_cost = 3.0;
+
+    // Cost of climbing terrain. INFINITY if we can't
+    float climb_cost = INFINITY;
+
+    // Cost of moving through a trap tile, INFINITY to avoid completely
+    float trap_cost = 0.;
+
+    // Cost of opening a door. INFINITY to never open doors, otherwise 2 to open and then move in.
+    float door_open_cost = INFINITY;
+
+    // Extra penalty for moving through rough terrain
+    float rough_terrain_cost = 0.;
+
+    // Extra penalty for moving through sharp terrain
+    float sharp_terrain_cost = 0.;
+
+    // Cost of moving up/down from any position. INFINITY if we can't fly.
+    float fly_cost = INFINITY;
+
+    // Cost of climbing stairs up and down. INFINITY if we can't.
+    float stair_movement_cost = INFINITY;
+
+    // If a mob is in the way currently, add this extra cost. INFINITY to always path around other critters.
+    float mob_presence_penalty = 0.;
+
+    PathfindingSettings() = default;
+    PathfindingSettings( PathfindingSettings const & ) = default;
+
+    bool operator==( const PathfindingSettings &rhs ) const = default;
```

Comment:
```suggestion
    auto operator<=>( const PathfindingSettings &rhs ) const = default;
```

removing default constructors would let consumers use [more sugary syntax](https://en.cppreference.com/w/cpp/language/aggregate_initialization):

```c++
const auto path_settings = PathfindingSettings{
    .bash_strength = bash_skill(),
    .mob_presence_penalty = 10.0,
};
```